lua/cmp_fuzzy_path/init.lua

```diff
@@ -12,8 +12,24 @@ local source = {
 local defaults = {
   fd_cmd = { 'fd', '-d', '20', '-p', '-i' },
   fd_timeout_msec = 500,
+  cmd_trigger_regex = [[^\%(e\|w\)\s\+]],
+  path_regex = [[^\%(\k\?[/:\~]\+\|\.\?\.\/\)\S\+]],
 }
 
+---Cache compiled regexes
+---@type table<string, userdata>
+local compiled = {}
+
+---Return a compiled regex from the cache, or create a new one and cache it
+---@param pattern string
+---@return userdata
+local function regex(pattern)
+  if not compiled[pattern] then
```

Comment:
Sorry for nitpicking.. 
The fast path should not need two lookups.
change to something like this:
```lua
local result = compiled[pattern]
if result then
   return result
end
compiled[pattern] = vim.regex(pattern)
return compiled[pattern]
```

Reply:
![image](https://github.com/tzachar/cmp-fuzzy-path/assets/76579810/2873b15e-938f-48e8-941e-e0d83c2de964)
 
![image](https://github.com/tzachar/cmp-fuzzy-path/assets/76579810/73b01a70-4c2e-4b0b-9150-61448e9afdf6)

![image](https://github.com/tzachar/cmp-fuzzy-path/assets/76579810/20657398-6d56-4cd9-aa53-33fe46ea30bf)

![image](https://github.com/tzachar/cmp-fuzzy-path/assets/76579810/a70a6183-31d9-40fc-bc42-8e41098cbed6)


@tzachar Actually I doubt if this really promotes the performance in a significant way :/ Tested many times on my machine and the time taken is very close.

For reference I will provide the perf code below:

```lua
local pattern = [[^\%(\k\?[/:\~]\+\|\.\?\.\/\)\S\+]]
local compiled = {}
local compiled_fast = {}

local function regex_fast(regex_pattern)
  local result = compiled_fast[regex_pattern]
  if result then
    return result
  end
  compiled_fast[regex_pattern] = vim.regex(regex_pattern)
  return compiled_fast[regex_pattern]
end

local function regex(regex_pattern)
  if not compiled[regex_pattern] then
    compiled[regex_pattern] = vim.regex(regex_pattern)
  end
  return compiled[regex_pattern]
end

local function time(fn, ...)
  local start = vim.loop.hrtime()
  for _ = 1, 1000 do
    fn(...)
  end
  return vim.loop.hrtime() - start
end

local function benchmark(epochs, name, fn, ...)
  local total = 0
  for _ = 1, epochs do
      total = total + time(fn, ...)
  end
  print(string.format('Average time of %s: %f ns', name, total / epochs))
end

benchmark(100, 'fast get regex', regex_fast, pattern)
benchmark(100, 'get regex', regex, pattern)
```